gpii/node_modules/ontologyHandler/src/ontologyHandler.js

```diff
@@ -51,21 +53,42 @@ URLs - this aspect is taken care of in the preferences server component.
         listeners: {
             onCreate: "gpii.ontologyHandler.loadOntologies"
         },
-        ontologySourceListDir: "%universal/testData/ontologies", // directory with ontologies
+        ontologyTransformationListFolder: "%universal/testData/ontologies/mappings", // directory with ontologies
+        ontologyMetadataFolder: "%universal/testData/ontologies", // directory with ontologies
         ontologyTransformSpecs: "", // will hold the actual ontology transformation definitions after read
-        ontologyTransformRoutes: "" // will hold results of Floyd-Warshall for generating paths between specs
+        ontologyTransformRoutes: "", // will hold results of Floyd-Warshall for generating paths between specs
+        ontologyMetadata: {} // will hold the actual ontolgoy metadata after files have been read
     });
 
     gpii.ontologyHandler.loadOntologies = function (that) {
         // Check for re-occurence of GPII-1063
         fluid.log("loadOntologies for " + that.id + " path " + fluid.pathForComponent(that));
-        var ontologyDir = fluid.module.resolvePath(that.options.ontologySourceListDir);
 
-        // Read the list of available ontologyTransformations
-        var ontologyTransformsList = fs.readdirSync(ontologyDir);
-        // Load all the ontologies:
-        gpii.ontologyHandler.loadOntologyTransformSpecs(that, ontologyDir, ontologyTransformsList);
+        // Load ontology transformations
+        var transformsFolder = fluid.module.resolvePath(that.options.ontologyTransformationListFolder);
+        var ontologyTransformsList = fs.readdirSync(transformsFolder); // Read the list of available ontologyTransformations
+        // Load the ontologies:
+        gpii.ontologyHandler.loadOntologyTransformSpecs(that, transformsFolder, ontologyTransformsList);
+
+        // Load the ontology metadata:
+        var metadataFolder = fluid.module.resolvePath(that.options.ontologyMetadataFolder);
+        gpii.ontologyHandler.loadOntologyMetadata(that, metadataFolder);
+    };
+
+    gpii.ontologyHandler.loadOntologyMetadata = function (that, metadataFolder) {
+        var metadataList = fs.readdirSync(metadataFolder);
+        fluid.each(metadataList, function (filename) {
+            // dont attempt to read the "mappings" folder:
+            if (filename === "mappings") {
```

Comment:
Less of a hack is to use fs.stat and just filter out all directories - http://stackoverflow.com/questions/15630770/node-js-check-if-path-is-file-or-directory